src/dune_rules/ctypes_stanza.ml

```diff
@@ -0,0 +1,141 @@
+open! Stdune
+open Import
+open Dune_lang.Decoder
+
+module Build_flags_resolver = struct
+
+  module Vendored = struct
+    type t =
+      { c_flags : Ordered_set_lang.Unexpanded.t
+      ; c_library_flags : Ordered_set_lang.Unexpanded.t }
+
+    let decode =
+      fields
+        (let+ c_flags = Ordered_set_lang.Unexpanded.field "c_flags"
+          and+ c_library_flags = Ordered_set_lang.Unexpanded.field "c_library_flags" in
+          { c_flags; c_library_flags })
+  end
+
+  type t =
+    | Pkg_config
+    | Vendored of Vendored.t
```

Comment:
The name Vendored isn't great here. The library could be in source, or it could be external to the workspace. What's important is that the user is providing the flags. So perhaps something like `custom` is better?